src/coupling.jl

```diff
@@ -18,10 +18,11 @@ end
 
 function cat_problems(prob::DiscreteProblem,prob_control::AbstractODEProblem)
   l = length(prob.u0) # add l_c = length(prob_control.u0)
+  if !(typeof(prob.f) <: typeof(DiffEqBase.DISCRETE_INPLACE_DEFAULT))
+    warn("Coupling to DiscreteProblem with nontrivial f.")
```

Comment:
Something like `""Coupling to DiscreteProblem with nontrivial f. Note that, unless the `f` was intended to be solved with `scale_by_time=true`, then the meaning of `f` will change when using an ODE/SDE/DDE/DAE solver."